pkg/server/sessions/manager.go

```diff
@@ -65,30 +63,38 @@ func NewManager(options *Options) (*manager, error) {
 	if options == nil {
 		return nil, ErrInvalidOptionsProvided
 	}
-	if options.MaxSessionAgeTime == 0 {
-		options.MaxSessionAgeTime = infinity
-	}
-	if options.MaxSessionInactivityTime == 0 {
-		options.MaxSessionInactivityTime = infinity
-	}
-	if options.Timeout == 0 {
-		options.Timeout = infinity
+
+	err := options.Validate()
+	if err != nil {
+		return nil, err
 	}
+
 	guard := &manager{
 		sessions: make(map[string]*Session),
 		ticker:   time.NewTicker(options.SessionGuardCheckInterval),
 		done:     make(chan bool),
 		logger:   logger.NewSimpleLogger("immudb session guard", os.Stdout),
-		options:  options,
+		options:  *options,
+	}
+
+	if guard.options.MaxSessionAgeTime == 0 {
```

Comment:
this settings may be moved into options struct

Reply:
Fixed in a599c8d448c9b98bf8e2a9d451ebba9a3f38574e